Add Mautic 3 upgrade stats, fix chart layout, add Travis CI

dennisameling opened the pull request:
This PR will do the following:
- [x] Add endpoint for gathering Mautic 3 upgrade telemetry (3 statuses: `started`, `failed`, `succeeded`), so we can see the % of successful upgrades. Endpoint is `POST mautic3upgrade/send`, so it will be `POST https://updates.mautic.org/stats/mautic3upgrade/send`.
- [x] Remove dependency on `babdev/mautic-bundle` as it doesn't exist anymore (and the Mautic instance it was using for tracking doesn't exist anymore)
- [x] Fix chart layout (has been broken for a while), also updates ChartJS to 2.9.3
- [x] Add Travis CI
- [x] Test if this works as expected with the [new upgrade script](https://github.com/mautic/mautic/issues/8819)

After deploying this PR on production (via FTP/SSH, I can do that after this PR is merged), due to the lack of DB migrations, the following needs to be done:
- `bin/console doctrine:schema:update --dump-sql`
- Check the SQL output to verify if we're not messing with the existing `Stats` table in the DB
- `bin/console doctrine:schema:update --force` to create the new `Mautic3UpgradeStats` table
- `bin/console cache:clear`

dennisameling committed:
Add Mautic 3 upgrade stats

dennisameling committed:
Add errorCode property for M3 upgrade

dennisameling committed:
Fix charts, add m3 upgrade dashboard

dennisameling commented:
Fixed the charts as well with a basic Bootstrap layout + upgraded ChartJS from 1 to 2.9.3

![image](https://user-images.githubusercontent.com/17739158/84600449-91495700-ae79-11ea-8ca2-38d21f616eb7.png)

Added M3 upgrade statistics dashboard:

![image](https://user-images.githubusercontent.com/17739158/84601306-69f58880-ae7f-11ea-8ff3-ebdf9642e805.png)

dennisameling committed:
Fix case where no entries with error codes exist

dennisameling committed:
Add tests

dennisameling committed:
Add Travis config

dennisameling committed:
Add empty parameters.yml.dist for Travis

dennisameling committed:
First copy parameters.yml.dist in Travis

dennisameling committed:
add dummy secret for Travis

dennisameling committed:
Remove upgradeStatus rule as it's not necessary

escopecz reviewed (commented):
There is some code I'd normally point out. Like that there is way too much logic in the controller, some comments ([like this one](https://github.com/mautic/statsapp/pull/1/files#diff-e209436ae06a2408e6b3c28dd6147bc2R15)) are not correct and/or needed. And that we should drop PHP 5.6. But this is not the core product. Just a side project so I won't require the changes. I think it's awesome that you did these changes as no one dared to touch this project for years. However, I'd feel more confident to approve this if the new POST endpoint was covered with functional tests.

dennisameling committed:
Add test for Mautic 3 upgrade POST endpoint

dennisameling committed:
Merge branch 'add-mautic-3-upgrade-stats' of https://github.com/dennisameling/statsapp into add-mautic-3-upgrade-stats

dennisameling commented:
@escopecz Thanks for checking! Agree that there's quite some room for improvements, but needed to get this done for the M3 release so had quite some time pressure. Also can't drop 5.6 support yet, as the server this is running on still runs on 5.6. I can upgrade that, but want to do that (soon) after the M3 release to have enough time for thoroughly testing everything.

I just added the test for the POST endpoint (sorry, forgot about that) and removed the comment (copied that from the other Entity). 

If you come across anything that you think needs to be fixed (e.g. moving logic from controller into a separate service), please feel free to leave a comment here, and I'll make sure to work on that sometime in the coming weeks. Also want to upgrade dependencies like Symfony so that it's easier to support this repo.
